### PR TITLE
Improved tests for statestore - redis

### PR DESCRIPTION
**Pull request by akremsa**
**What this PR does / Why we need it**:
I've noticed that redis.go has low unit test coverage and decided to improve that.

Before:
```
go test -coverprofile=before.out
PASS
coverage: 52.3% of statements
ok  	open-match.dev/open-match/internal/statestore	1.394s
```

After:
```
go test -coverprofile=after.out
PASS
coverage: 84.8% of statements
ok  	open-match.dev/open-match/internal/statestore	1.007s
```

### CODE REVIEW

File: internal/statestore/redis_test.go [hunk 1]
```diff
@@ -247,6 +357,515 @@ func TestConnect(t *testing.T) {
...
+
+	// Verify all tickets are created and returned
+	verifyTickets(service, tickets)
+
+	c, err := redis.Dial("tcp", fmt.Sprintf("%s:%s", cfg.GetString("redis.hostname"), cfg.GetString("redis.port")))
+	require.NoError(t, err)
+	// Add the first ticket to the pending release and verify changes are reflected in the result
+	redis.Strings(c.Do("ZADD", "proposed_ticket_ids", time.Now().UnixNano(), ids[0]))
+
+	// Verrify 1 ticket is indexed
```
Comment (aLekSer) [inline internal/statestore/redis_test.go:756]:
```suggestion
	// Verify 1 ticket is indexed
```

Comment (akremsa) [reply internal/statestore/redis_test.go]: Thanks

File: internal/statestore/redis_test.go [hunk 2]
```diff
@@ -247,6 +357,515 @@ func TestConnect(t *testing.T) {
...
+			description:     "empty id passed, err expected",
+			ticketID:        "",
+			expectedCode:    codes.NotFound,
+			expectedMessage: "Ticket id: not found",
+		},
+		{
+			description:     "wrong id passed, err expected",
+			ticketID:        "123456",
+			expectedCode:    codes.NotFound,
+			expectedMessage: "Ticket id:123456 not found",
```
Comment (aLekSer) [inline internal/statestore/redis_test.go:479]: Do we need a whitespace, but that could do this in a separate PR.
```suggestion
			expectedMessage: "Ticket id: 123456 not found",
```

**Comment by akremsa** [comment]
@Laremere could you please take a look?

File: internal/statestore/redis_test.go (hunk 2 above)
Comment (akremsa) [reply internal/statestore/redis_test.go]: ok

**Review by aLekSer — APPROVED** [review]
Fine with me.

**Review by Laremere — APPROVED** [review]
There was already reasonably good coverage of this through the e2e tests (unfortunately coverage doesn't work across packages.)  This seems like enough of a positive, and don't want to waste your work, so approving.
I think in the future if there are instances where these tests need to change and coverage is handled by e2e tests, I'll be fine with PR authors either fixing this test, or removing test cases here (again, as long as it's also covered well by e2e.)

**Comment by Laremere** [comment]
Thanks for the PR!
